optimizely/event/user_event_factory.py

```diff
@@ -36,10 +39,13 @@ def create_impression_event(
       - activated_experiment is None.
     """
 
-        if not activated_experiment:
+        if not activated_experiment and flag_type is not enums.DecisionSources.ROLLOUT:
             return None
 
-        experiment_key = activated_experiment.key
+        if activated_experiment:
+            experiment_key = activated_experiment.key
+        else:
+            experiment_key = None
         variation = project_config.get_variation_from_id(experiment_key, variation_id)
```

Comment:
How will this succeed if `experiment_key` is `None`? It will generate errors logs based on: https://github.com/optimizely/python-sdk/blob/master/optimizely/project_config.py#L340-L364.